[Ronald Tan Sing Wei] iP

RonaldTanSingWei opened the pull request:
# DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ *SUPER* FAST to use

All you need to do is.
1. Download it from [here](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/project.html),
2. double-click it,
3. add your tasks,
4. let it manage your tasks for you :wink:

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon!)
- [ ] Reminders (coming soon!)
#

If you Java programmer, you can use it to practice Java too. Here's the `main` method: 
```javascript
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Commit:
Increment 1, added to Duke.java

Commit:
Increment Two, added to Duke.java, need to fix bug where space causes two items to be added to list instead of one

Commit:
Increment 2, added to Duke.java, changed hasNext to hasNextLine and scanner input from next() to nextLine()

Commit:
Increment 3, added new Task class

Commit:
Increment 3, updated Task.java

Commit:
Increment 2: adding list numbers to Duke.java

Commit:
Increment 3 progress

Commit:
Increment 3 completed in Duke.java

Commit:
Increment 4 Initial work done on Duke and Task

Commit:
Increment 4, created new types of Task extending Task

Commit:
Increment 4 (remaining bug when event or deadline task does not have "/" character)

Commit:
Increment 5

Commit:
Increment 5, accounted for some invalid inputs

Commit:
Increment 5, adjusted for ArrayList and invalid inputs

Commit:
Increment 6, added delete

Commit:
Level 7 commit

Commit:
Update level-7

Commit:
Implement level-7

Added the ability to read files and save them to store the list created for future usage.

Commit:
Add level-8

Commit:
Update level-7 increment

Commit:
Update level-7 increment

Commit:
Add Level-8-increment

LocalDate and DateTimeFormatter are imported to allow program to check if user input of a date is correct, and add it to the list accordingly

Commit:
Merge branch 'branch-Level-7' into branch-Level-8

# Conflicts:
#	src/main/java/Duke.java
Resolve conflicts

Commit:
A-More OOP increment
Added class files

Commit:
Update OOP Increment

Commit:
Finish OOP Increment

Separated Duke.java into multiple classes to handle different processes, for easier delegation and management of inputs

Commit:
Tidy OOP increment

Commit:
Add A-Packages

Placed all DUKE related files in a DUKE package for easier management

Commit:
Add JUnit Tests

Created TaskTest file to use for testing if Task class works and TaskList class works

Commit:
Update JUnit increment

Commit:
Add A-JavaDoc increment

Updated files to include comments

Commit:
Add A-JavaDoc

Comments that detail each class's functions and their important methods are provided

Commit:
Add A-CodingStandard

Adjusted code to be neater

Commit:
Add Level-9 increment

Added find method to allow users to find keywords in their list

Commit:
Merge tag 'A-CodingStandard'

Commit:
Merge tag 'A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Storage.java

Commit:
Update Storage to be more concise

aryagiat commented on src/main/java/duke/DeadlineTask.java:
```diff
@@ -0,0 +1,25 @@
+    protected String type;
+    protected String time;
+
+    public DeadlineTask(String description, String time) {
```
Should there be javadocs comments for public methods?

aryagiat commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,74 @@
+ * The Duke class acts as a text bot that records down the tasks given to it. It can
+ * then mark these tasks as completed and delete them based on the inputs given.
+ */
+
```
Is this space needed?

aryagiat commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,74 @@
+ */
+
+public class Duke {
+    /**
+     * Checks the date input from the user to see if it is a valid date
+     *
+     * @param dateStr Date taken in to check for validity
+     * @return If date is valid
+     */
```
Is this Javadoc located correctly? Should it be on line 34 instead?

aryagiat commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,74 @@
+        tasks = new TaskList(storage.loadFile());
+    }
+
+    public void run() {
```
Should there be javadoc comments for public classes/methods?

aryagiat commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,182 @@
+     * @param ui Ui to close when necessary
+     * @param tasklist The list that is adjusted when a specific input is given
+     */
+    public static void parse(String in, Ui ui, TaskList tasklist) {
```
Should a parser be responsible in performing commands or should it just parse user input such that it can be meaningful to the program?

aryagiat commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,90 @@
+    public ArrayList loadFile() {
+        ArrayList<Task> storedList = new ArrayList<Task>();
+        try {
+            File taskList = new File("C:\\Users\\ronal\\OneDrive\\Desktop\\CS2103\\DUKE\\data\\duke.txt");
```
Should this be a relative path instead?

aryagiat commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,143 @@
+
+    /**
+     * Delets a task from the tasklist
+     * @param i
```
what is the description for parameter i?

Commit:
Merge branch 'add-gradle-support' of https://github.com/RonaldTanSingWei/ip

Commit:
Implement GUI

Create files for DUKE to use GUI

Commit:
Update Duke to be more GUI friendly

Commit:
Add level 10

Update GUI to store tasks and close smoothly

aryagiat commented on src/main/java/MainWindow.java:
```diff
@@ -0,0 +1,53 @@
+import duke.Duke;
```
Should this file belong to a package?

ryanpeh commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,90 @@
+ */
+public class Storage {
+
+    public Storage (String filePath) {
```
Perhaps you could remove the parameter `String filePath` as it seems like it is not used anywhere else in the code?

ryanpeh replied:
Also I believe that there shouldn't be any spaces before the parenthesis?

ryanpeh commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,89 @@
+import java.io.File;
+import java.io.FileWriter;
+
+import static java.lang.Integer.parseInt;
```
Should these imports be removed? Only `java.time.LocalDate` is used.

ryanpeh commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,90 @@
+                } else if (item.charAt(1) == 'D') {
+                    int i = item.indexOf("(");
+                    DeadlineTask addedTask = new DeadlineTask(item.substring(7, i),
+                            item.substring(i+5, item.length()-1));
+                    if (item.charAt(4) == 'X') {
+                        addedTask.markAsDone();
+                    }
+                    storedList.add(addedTask);
+                } else if (item.charAt(1) == 'E') {
+                    int i = item.indexOf("(");
+                    EventTask addedTask = new EventTask(item.substring(7, i),
+                            item.substring(i+5, item.length()-1));
```
Perhaps there could be spaces around the operators?

ryanpeh commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,90 @@
+        try {
+            FileWriter listEditor = new FileWriter("./data/duke.txt");
+            for (Task item: taskList.tasks) {
+                if(item != null) {
```
Perhaps the `if` could be followed by a whitespace?

ryanpeh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,182 @@
+package duke;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import static java.lang.Integer.parseInt;
```
Should the imports be in this order instead?
``` java
import java.time.LocalDate;
import java.time.format.DateTimeFormatter;
import java.time.format.DateTimeParseException;

import static java.lang.Integer.parseInt;
```

ryanpeh commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,150 @@
+ */
+public class TaskList {
+    public ArrayList<Task> tasks;
+    public int count;
```
Perhaps these class variables should have a `private` access modifier instead?

ryanpeh commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,150 @@
+    public String addToDoTask(String in) {
+        String output = "Got it. I've added this task: \n";
+        tasks.add(new ToDoTask(in.substring(5)));
+        output += tasks.get(tasks.size()-1) + "\n";
```
Perhaps there could be spaces around the operators?

ryanpeh commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,20 @@
+
+public class Ui {
+
+    public Scanner input;
```
Perhaps this class variable should have a `private` access modifier instead?

ryanpeh commented on src/main/java/duke/ToDoTask.java:
```diff
@@ -0,0 +1,20 @@
+/**
+ * A subclass of Task that can be stored inside the tasklist
+ */
+public class ToDoTask extends Task{
```
Should there be a whitespace before the curly bracket?

ryanpeh commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,150 @@
+        this.tasks = tasks;
+        this.count = tasks.size();
+    }
+
```
Perhaps it's good to have javadocs for constructors.

ryanpeh commented on src/main/java/Launcher.java:
```diff
@@ -0,0 +1,9 @@
+import duke.Duke;
```
Should this import be removed?

ryanpeh commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,90 @@
+package duke;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.util.ArrayList;
+import java.util.Scanner;
```
Perhaps the separation in import group can be removed?

ryanpeh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,182 @@
+                return Parser.eventOutput(in, tasklist);
+            } else if (in.substring(0, 8).equals("deadline")) {
+                return Parser.deadlineOutput(in, tasklist);
+            } else if (in.substring(0,4).equals("find")) {
```
Perhaps there should be a whitespace after the comma?

ryanpeh commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,129 @@
+        dialogContainer.heightProperty().addListener((observable) -> scrollPane.setVvalue(1.0));
+
+        sendButton.setOnMouseClicked((event) -> {
+                handleUserInput();
+        });
+
+        userInput.setOnAction((event) -> {
+                handleUserInput();
```
Perhaps the indentation of `handleUserInput()` should be 4 spaces instead of 8?

ryanpeh commented on src/main/java/duke/EventTask.java:
```diff
@@ -0,0 +1,25 @@
+/**
+ * A subclass of Task that can be stored inside the tasklist
+ */
+public class EventTask extends Task{
```
Perhaps there should be a whitespace before the curly bracket?

ryanpeh commented on src/main/java/duke/DeadlineTask.java:
```diff
@@ -0,0 +1,25 @@
+/**
+ * A subclass of Task that can be stored inside the tasklist
+ */
+public class DeadlineTask extends Task{
```
Should there be a whitespace before the curly bracket?

ryanpeh commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,89 @@
+        tasks = new TaskList(storage.loadFile());
+    }
+/*
+    public void run() {
+
+        Scanner input = new Scanner(System.in);
+        ui.input = input;
+        try {
+            while (input.hasNextLine()) {
+                String userInput = ui.readInput();
+                Parser.parse(userInput, this.ui, this.tasks);
+            }
+        } catch (IllegalStateException e) {
+            storage.saveFile(tasks);
+        }
+    }
+
+    public static void main(String[] args) {
+        new Duke("./data/duke.txt").run();
+    }*/
```
Should this commented out code be removed?

Commit:
Format code to be more standardized

Code missing white spaces where necessary, need to tidy code to be more readable

Input whitespaces and deleted unnecessary code and comments

This allows code to be more readable for viewers as it follows the standard conventions

Commit:
Format code to be more standardized

Code missing white spaces where necessary, need to tidy code to be more readable

Input whitespaces and deleted unnecessary code and comments

This allows code to be more readable for viewers as it follows the standard conventions

Commit:
Implement Assertions into Duke

Usage of many if-else statements to check for valid strings, can be hard to read

Replace these if-else staements with assert statements

This allows code to be more readable and less likely to have mistakes due to wrong implementation of if-else statements

RonaldTanSingWei committed:
Merge pull request #1 from RonaldTanSingWei/branch-Assertions

Implement Assertions into Duke

Commit:
Merge branch 'master' of https://github.com/RonaldTanSingWei/ip into branch-CodeQuality

Commit:
Merge CodeQuality into Master Branch

Commit:
Implement Reschedule method

Currently, rescheduling a task requires the user to delete and add the same task with a new date, which changes the order of the tasklist

This is very inconvenient for some users

As such, the reschedule method allows users to reschedule the date easily

This allows them to schedule tasks to a different date easily, while preserving original order

Commit:
Implement Reschedule method

Currently, rescheduling a task requires the user to delete and add the same task with a new date, which changes the order of the tasklist

This is very inconvenient for some users

As such, the reschedule method allows users to reschedule the date easily

This allows them to schedule tasks to a different date easily, while preserving original order

Commit:
File movements

Commit:
Add user guide

Commit:
Changes to Duke
